sklearn/neural_network/tests/test_mlp.py

```diff
@@ -869,3 +870,30 @@ def test_mlp_param_dtypes(dtype, Estimator):
 
     if Estimator == MLPRegressor:
         assert pred.dtype == dtype
+
+
+def test_mlp_loading_from_joblib_partial_fit(tmp_path):
+    """Loading from MLP and partial fitting updates weights. Non-regression
+    test for #19626."""
+    pre_trained_estimator = MLPRegressor(
+        hidden_layer_sizes=(42,), random_state=42, learning_rate_init=0.01, max_iter=200
+    )
+    features, target = [[2]], [4]
+
+    # Fit on x=2, y=4
+    pre_trained_estimator.fit(features, target)
+
+    # dump and load model
+    pickled_file = tmp_path / "mlp.pkl"
+    joblib.dump(pre_trained_estimator, pickled_file)
+    load_estimator = joblib.load(pickled_file)
+
+    # Train for a more epochs on point x=2, y=1
+    fine_tune_features, fine_tune_target = [[2]], [1]
+
+    for _ in range(200):
+        load_estimator.partial_fit(fine_tune_features, fine_tune_target)
+
+    # finetuned model learned the new target
+    predicted_value = load_estimator.predict(fine_tune_features)
+    assert_allclose(predicted_value, fine_tune_target, rtol=1e-4)
```

Comment:
Do you think that it might be nice to have a test to check the equality of `id` of `coefs_` and `intercepts_` between 2 `partial_fit` to be sure that in the future, one does not try to make a copy?

Reply:
Checking for ids between two partial fits still passes on `main`. If we were to have this test, it should check that the `coefs_` in MLP are the same objects in `BaseOptimizer`. Since this PR removes params in `BaseOptimizer`, there is nothing left to check. `MLP` is the only object referencing the params.

Reply:
OK